code/game/objects/items/weapons/traps.dm

```diff
@@ -123,10 +124,519 @@
+/obj/item/weapon/trap/update_icon()
+	icon_state = "[icon_base][deployed]"
+
+/obj/item/weapon/trap/animal
+	name = "small trap"
+	throw_speed = 2
+	throw_range = 1
+	gender = PLURAL
+	icon = 'icons/obj/items.dmi'
+	icon_base = "small"
+	icon_state = "small0"
+	desc = "A small mechanical trap thas is used to catch small animals like mice, lizards, chick and spiderlings."
+	throwforce = 2
+	force = 1
+	w_class = 2
+	origin_tech = list(TECH_MATERIAL = 1)
+	matter = list(DEFAULT_WALL_MATERIAL = 1750)
+	deployed = 0
+	time_to_escape = 3 // Minutes
+	var/breakout = FALSE
+	var/last_shake = 0
+	var/list/allowed_mobs = list(/mob/living/simple_animal/mouse, /mob/living/simple_animal/chick, /mob/living/simple_animal/lizard)
+	var/release_time = 0
+	var/list/resources = list(rods = 6)
+
+/obj/item/weapon/trap/animal/examine(mob/user)
 	..()
+	if(contents.len && isliving(contents[1]))
+		var/message = "<span class='notice'>\The [src] has [contents[1].name] and it is "
+		if(contents[1].stat == DEAD)
+			message += "<span class='danger'>dead</span>"
+		else if(contents[1].stat == UNCONSCIOUS)
+			message += "<span class='warning'>unconscious</span>"
+		else if((contents[1].maxHealth / contents[1].health) == 1)
+			message += "<span class='good'>healthy</span>"
+		else
+			message += "<span class='warning'>wounded</span>"
+		message += ".</span>"
+		to_chat(user, message)
+	else if(contents.len) // spiderling
+		to_chat(user, "<span class='notice'>\The [src] has [contents[1].name] and it is alive.</span>")
+	else
+		to_chat(user, "<span class='notice'>\The [src] is empty.</span>")
+
+/obj/item/weapon/trap/animal/Crossed(AM as mob|obj)
+	if(world.time - release_time < 50) // If we just released the animal, not to let it get caught again right away
+		return
+
+	if(contents.len) // It is full
+		return
+
+	if(isliving(AM))
+		var/mob/living/L = AM
+		if(L.isMonkey() && (/mob/living/carbon/human/monkey in allowed_mobs)) // Because monkeys can be of type of just human.
+			L.visible_message(
+				"<span class='danger'>[L] enters \the [src], and it snaps shut with a clatter!</span>",
+				"<span class='danger'>You enters \the [src], and it snaps shut with a clatter!</span>",
+				"<b>You hear a loud metallic snap!</b>"
+				)
+			L.forceMove(src)
+			playsound(src, 'sound/weapons/beartrap_shut.ogg', 100, 1)
+			deployed = 1
+			update_icon()
+			src.animate_shake()
+			return
+		for(var/f in allowed_mobs)
+			if(istype(AM, f))
+				L.visible_message(
+					"<span class='danger'>[L] enters \the [src], and it snaps shut with a clatter!</span>",
+					"<span class='danger'>You enters \the [src], and it snaps shut with a clatter!</span>",
+					"<b>You hear a loud metallic snap!</b>"
+					)
+				L.forceMove(src)
+				playsound(src, 'sound/weapons/beartrap_shut.ogg', 100, 1)
+				deployed = 1
+				update_icon()
+				src.animate_shake()
+
+	else if(istype(AM, /obj/effect/spider/spiderling)) // for spiderlings
+		var/obj/effect/spider/spiderling/S = AM
```

Comment:
Code Duplication - Capture
Extract all of them into a function with a optional parameter to stop processing.